src/Clash/Cores/Ethernet/Examples/TxStack.hs

```diff
@@ -1,49 +1,162 @@
-{-# language FlexibleContexts #-}
-
 {-|
-Module      : Clash.Cores.Ethernet.Examples.TxStack
-Description : Provides the entire transmit stack as a circuit.
+  Module      : Clash.Cores.Ethernet.Examples.TxStack
+  Copyright   : (C) 2024, Tim Wallet <tw.personal@tutanota.com>
+  Description : Provides a standard Ethernet MAC transmit stack.
+  License     : BSD2 (see the file LICENSE)
+  Maintainer  : QBayLogic B.V. <devops@qbaylogic.com>
+
+A fully modular MAC transmit stack which allows the transmission of data over Ethernet II.
+Supports any data width bigger than zero.
+
+Example usage:
+
+>>> :set -XFlexibleContexts
+>>> :set -XMultiParamTypeClasses
+>>> import Clash.Prelude
+>>> import Clash.Cores.Crc
+>>> import Clash.Cores.Crc.Catalog
+>>> import Clash.Cores.Ethernet.Mac.EthernetTypes
+>>> import Data.Proxy
+>>> import Protocols
+>>> import Protocols.Extra.PacketStream
+
+The Ethernet TX PHY is completely interchangeable with this stack. In the example below,
+we use a dummy. You have to replace this dummy variable with an Ethernet TX PHY circuit
+for your specific hardware (e.g. RGMII, MII or SGMII) that is adapted to the
+`PacketStream` protocol, i.e. with type:
+
+>>> :{
+dummyTxPhy
+  :: HiddenClockResetEnable domEthTx
+  => Circuit (PacketStream domEthTx 1 ()) (PacketStream domEthTx 1 ())
+dummyTxPhy = undefined
+:}
+
+For example, the Lattice ECP5 board uses an RGMII PHY, found at `Clash.Lattice.ECP5.RGMII.rgmiiTxC`.
+
+`txStack` is the most common ethernet MAC TX stack that will be sufficient for
+most people. That is, it inserts an interpacket gap of 12 bytes, pads the payload
+to 46 bytes and assumes that you process the bytes in a different clock domain than the
+ethernet TX domain. All you have to do is specify the data width (in this example 4),
+the clock domains, and the TX PHY you want to use.
+
+The stack uses `Clash.Cores.Crc.crcEngine` internally to calculate the frame check
+sequence of the Ethernet frame. To be able to use this component, we need to use
+`Clash.Cores.Crc.deriveHardwareCrc` to derive the necessary instance.
+
+>>> :{
+$(deriveHardwareCrc (Proxy @Crc32_ethernet) d8 d4)
```

Comment:
If you add a `deriveHardwareCrc` here it's not required to add the `HardwareCrc` constraint since it can resolve it locally.